client/state/purchases/selectors.js

```diff
@@ -21,7 +24,7 @@ export const getPurchasesError = state => state.purchases.error;
  * @return {Object} the matching purchase if there is one
  */
 export const getByPurchaseId = ( state, purchaseId ) => (
-	getPurchases( state ).filter( purchase => purchase.id === purchaseId ).shift()
+	purchasesAssembler.createPurchasesArray( getPurchases( state ).filter( purchase => purchase.ID === purchaseId ) ).shift()
```

Comment:
Should we assemble purchases in `getSitePurchases` and `getUserPurchases` as well?